Message-Id: <07010a2ad79559c412949f0005dbe3cb03d8416e.1498504812.git.ps@pks.im>
From: Patrick Steinhardt <ps@pks.im>
Date: Mon, 26 Jun 2017 21:20:45 +0200
Subject: [PATCH] 9pfs: include <linux/limits.h> for XATTR_SIZE_MAX

The function `v9fs_xattrcreate` makes use of the define `XATTR_SIZE_MAX`
to reject attempts of creating xattrs with an invalid size, which is
defined in <linux/limits.h>. On glibc-based systems, this header is
indirectly included via <limits.h>, <bits/posix1_lim.h>,
<bitts/local_lim.h>, but on other platforms this is not guaranteed due
to not being part of the POSIX standard. One examples are systems based
on musl libc, which do not include the <linux/limits.h> indirectly,
which leads to `XATTR_SIZE_MAX` being undefined.

Fix this error by directly include <linux/limits.h>. As the 9P fs code
is being Linux-based either way, we can simply do so without breaking
other platforms. This enables building 9pfs on musl-based systems.

Signed-off-by: Patrick Steinhardt <ps@pks.im>
[ Changes by AF
 - Apply to the QEMU tree inside of Xen
]
Signed-off-by: Alistair Francis <alistair.francis@xilinx.com>
---
Upstream Status: Pending review and acceptance

 tools/qemu-xen/hw/9pfs/9p.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/qemu-xen/hw/9pfs/9p.c b/tools/qemu-xen/hw/9pfs/9p.c
index 96d2683348..48cd558e96 100644
--- a/tools/qemu-xen/hw/9pfs/9p.c
+++ b/tools/qemu-xen/hw/9pfs/9p.c
@@ -13,6 +13,7 @@
 
 #include "qemu/osdep.h"
 #include <glib/gprintf.h>
+#include <linux/limits.h>
 #include "hw/virtio/virtio.h"
 #include "qapi/error.h"
 #include "qemu/error-report.h"
-- 
2.13.2
